infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java

```diff
@@ -67,49 +87,195 @@ public class LogReplicationConfigManager {
+        initLogReplicationVersionPlugin();
+        setupVersionTable();
+    }
+
+    private void initLogReplicationVersionPlugin() {
+        log.info("Plugin :: {}", pluginConfigFilePath);
+        LogReplicationPluginConfig config = new LogReplicationPluginConfig(pluginConfigFilePath);
+        File jar = new File(config.getStreamFetcherPluginJARPath());
+        try (URLClassLoader child = new URLClassLoader(new URL[]{jar.toURI().toURL()}, this.getClass().getClassLoader())) {
+            Class plugin = Class.forName(config.getStreamFetcherClassCanonicalName(), true, child);
+            logReplicationVersionAdapter = (ILogReplicationVersionAdapter) plugin.getDeclaredConstructor()
+                    .newInstance();
+        } catch (Exception e) {
+            log.error("Fatal error: Failed to get Log Replicatior Version Plugin", e);
+            throw new UnrecoverableCorfuError(e);
+        }
+    }
+
+    /**
+     * Read registry table's latest entries into memory for config synchronization.
+     *
+     * @return True if registry table address space changed and the in-memory entries are refreshed.
+     */
+    public boolean loadRegistryTableEntries() {
+        // If NON_ADDRESS is given, retrieve the latest registry table entries with retry.
+        try {
+            log.info("Sync in-memory registry table entries with retry.");
+            return IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try {
+                    StreamAddressSpace currentAddressSpace = rt.getSequencerView().getStreamAddressSpace(
+                            new StreamAddressRange(REGISTRY_TABLE_ID, Long.MAX_VALUE, Address.NON_ADDRESS));
+                    if (lastAddressSpace == null || !lastAddressSpace.equals(currentAddressSpace)) {
+                        lastAddressSpace = currentAddressSpace;
+                        CorfuTable<TableName, CorfuRecord<TableDescriptors, TableMetadata>> registryTable =
+                                rt.getTableRegistry().getRegistryTable();
+                        registryTableEntries = registryTable.entryStream().collect(Collectors.toList());
+                        return true;
+                    }
+                } catch (Exception e) {
+                    log.error("Exception caught when syncing with registry table, retry needed", e);
+                    throw new RetryNeededException();
+                }
+                return false;
+            }).setOptions(x -> x.setMaxRetryThreshold(Duration.ofSeconds(SYNC_THRESHOLD))).run();
+        } catch (InterruptedException e) {
+            log.error("Cannot sync with registry table, LR stopped", e);
+            throw new UnrecoverableCorfuInterruptedError(e);
+        }
+    }
+
+    /**
+     * Read registry table's entries up to the given timestamp into memory for config synchronization.
+     *
+     * @param timestamp the timestamp up to which registry table should be synced. By default, (NON_ADDRESS) the latest
```

Comment:
NON_ADDRESS refers to timestamp=-1 which will be considered invalid by the check in L171.  I think we can remove the check and also the comment about the default behavior.